Add bluetooth documentation (#79)
Eeems-Org/remarkable.guide: src/devel/device/bluetooth.rst

=========
Bluetooth
=========

> By default bluetooth is not configured to work. But the module exists, the kernel is built to use it.

Author: How about this edit:

```suggestion
This article is about the <strong>ReMarkable Paper Pro</strong>.

By default, bluetooth on the rMPP is not configured to work. A Kernel module does exist, and the kernel is built to use it.
```

Reviewer: I should add some guidance to the readme, but we use the full name instead of rM1/rM2/rMPP throughout the guide. It's also styalized as reMarkable and not ReMarkable.

Reviewer: It might just be worth saying something like:

> The reMarkable 1 and 2 do not have bluetooth. The reMarkable Paper Pro does have bluetooth hardware, but the driver is not loaded by default.

Reviewer: Did you forget to push a commit? I'm still seeing the old text here.

Author: Ah sorry I updated the suggestion 😂 .

One of my pet peeves is folks being too quick to action, when discussing. I guess that conversely means I can be slow to action.

> ====
> Bluetooth
> ====

Reviewer: ```suggestion
=========
Bluetooth
=========
```

> By default Bluetooth is not configured to work. But the module exists, the kernel is built to use it.

Reviewer: ```suggestion
By default bluetooth is not configured to work. But the module exists, the kernel is built to use it.
```
To keep consistent with lack of capitalization elsewhere in the document

> Enabling bluetooth
> ------------------------

Reviewer: ```suggestion
Enabling bluetooth
------------------
```

> .. code-block:: shell
> 
>   ssh root@10.11.99.1
>   root@imx8mm-ferrari:~# modprobe btnxpuart
>   root@imx8mm-ferrari:~# bluetoothctl
>   Agent registered
>   [CHG] Controller 24:FD:FA:01:6E:A3 Pairable: yes
>   [bluetooth]# show
>   Controller 24:FD:FA:01:6E:A3 (public)
>     Name: imx8mm-ferrari
>     Alias: imx8mm-ferrari
>     Class: 0x00000000
>     Powered: no
>     Discoverable: no
>     DiscoverableTimeout: 0x000000b4
>     Pairable: yes
>     UUID: Generic Attribute Profile (00001801-0000-1000-8000-00805f9b34fb)
>     UUID: Generic Access Profile    (00001800-0000-1000-8000-00805f9b34fb)
>     UUID: PnP Information           (00001200-0000-1000-8000-00805f9b34fb)
>     UUID: A/V Remote Control Target (0000110c-0000-1000-8000-00805f9b34fb)
>     UUID: A/V Remote Control        (0000110e-0000-1000-8000-00805f9b34fb)
>     UUID: Device Information        (0000180a-0000-1000-8000-00805f9b34fb)
>     Modalias: usb:v1D6Bp0246d0541
>     Discovering: no
>     Roles: central
>     Roles: peripheral
>   Advertising Features:
>     ActiveInstances: 0x00 (0)
>     SupportedInstances: 0x06 (6)
>     SupportedIncludes: tx-power
>     SupportedIncludes: appearance
>     SupportedIncludes: local-name
>     SupportedSecondaryChannels: 1M
>     SupportedSecondaryChannels: 2M
>     SupportedSecondaryChannels: Coded

Reviewer: ```suggestion
.. code-block:: shell

  modprobe btnxpuart
  bluetoothctl show
```

=========
Bluetooth
=========

The reMarkable 1 and 2 do not have bluetooth. The reMarkable Paper Pro does have bluetooth hardware, but the driver is not loaded by default.

Enabling bluetooth
------------------

.. code-block:: shell

  modprobe btnxpuart
  bluetoothctl show

> .. raw:: html
> 
>     <div class="warning">
>         ⚠️ Experimental. ⚠️
> 
> The following information is from individual user experimentation, where information is still being gathered 
> 
> .. raw:: html
> 
>     </div>

Reviewer: ```suggestion
:raw-html:`<div class="warning">⚠️ FIXME. ⚠️`

This page is just a stub that needs to be completed. You can `open a PR on the repo <https://github.com/Eeems-Org/remarkable.guide>`_ to add more content to the page.
:raw-html:`</div>`
```

The entire site is from individual user experimentation, and the device documentation itself is meant to provide lower level information that still needs to be gathered. If you are saying that this page isn't done, then the stub notice would be the correct thing to add here: https://github.com/Eeems-Org/remarkable.guide?tab=readme-ov-file#how-do-i-mark-a-page-as-a-stub-that-needs-further-content-written

Connecting a Device
-------------------

> .. code-block:: shell
> 
>   [bluetooth]# power on
>   Changing power on succeeded
>   [CHG] Controller 00:10:20:30:40:50 Powered: yes
>   
>   [bluetooth]# scan on
>   Discovery started
>   [CHG] Controller 00:10:20:30:40:50 Discovering: yes
>   [NEW] Device 00:12:34:56:78:90 device name
>   [CHG] Device 00:12:34:56:78:90 LegacyPairing: yes
>   
>   [bluetooth]# pair 00:12:34:56:78:90
>   Attempting to pair with 00:12:34:56:78:90
>   [CHG] Device 00:12:34:56:78:90 Connected: yes
>   [CHG] Device 00:12:34:56:78:90 Connected: no
>   [CHG] Device 00:12:34:56:78:90 Connected: yes
>   Request PIN code
>   [agent] Enter PIN code: 1234
>   [CHG] Device 00:12:34:56:78:90 Paired: yes
>   Pairing successful
>   [CHG] Device 00:12:34:56:78:90 Connected: no
>   
>   [bluetooth]# connect 00:12:34:56:78:90
>   Attempting to connect to 00:12:34:56:78:90
>   [CHG] Device 00:12:34:56:78:90 Connected: yes
>   Connection successful

Reviewer: I'm not a big fan of including command output like this in the article. Some users will read this and have some difficulty determining what commands we are saying to run. 

```suggestion
.. code-block:: shell

  bluetoothctl

 This will drop you into an interactive prompt for running bluetoothctl commands.
 
 .. code-block:: shell
   power on
   scan on
   # Wait for the mac address for the device you are interested to show up
   pair <mac-address>
   # You will be prompted to enter a PIN code for the device
   connect <mac-address>
   trust <mac-address>
```

I did some digging and there seems to be no good way to scan without actually running the interact prompt, which is rather odd.

External Links
==============

> - Arch-Wiki on Bluetooth
>    https://wiki.archlinux.org/title/Bluetooth
> - Arch-Wiki on Bluetooth section on Pairing
>    https://wiki.archlinux.org/title/Bluetooth#Pairing
> - Arch-Wiki on Bluetooth section on Configuration
>    https://wiki.archlinux.org/title/Bluetooth#Configuration

Reviewer: ```suggestion
- ArchWiki article on Bluetooth
   https://wiki.archlinux.org/title/Bluetooth
```

Having three links to the same page isn't something I'd like to add, and we do try to stylize things the way they are intended by the authors.

- Kernel Module documentation in the i.MX knowledge base
   https://community.nxp.com/t5/i-MX-Processors-Knowledge-Base/Bluetooth-NXP-UART-Driver-Linux-BSP-6-1-22-btnxpuart/ta-p/1708588

> - Bluez (bluetoothctl) homepage
>    https://www.bluez.org/

Reviewer: ```suggestion
- Documentation for bluetoothctl and related commands
   https://man.archlinux.org/listing/extra/bluez-utils/
```
This isn't really linking someone to something that immediately has useful information, they instead have to do a bunch of digging to find any useful documentation on bluetoothctl. I would much rather link to hosted documentation for bluetoothctl and other commands bluez provides.
